crates/cxx-qt-gen/src/parser/signals.rs

```diff
@@ -110,6 +132,30 @@ impl ParsedSignal {
             docs,
         })
     }
+
+    /// Returns a [TokenStream] of the signals parameters, which can be used in generation stage
+    /// class_name is the rust class name which this method is defined for
+    pub fn get_params_tokens(&self, class_name: &Ident) -> TokenStream {
```

Comment:
assume we can't sanely shared code with the one in parsed method?

Guess this is what you were saying about traits? If ParsedMethod and ParsedSignal implement a ParamTokens trait or something that had methods to get the mutable / parameters etc. Then the generation phase could have one method of `get_param_token` for a type that `impl ParamTokens` or something?

Reply:
@LeonMatthes any thoughts here if it's worth making a trait to have a common way for ParsedSignal/ParsedMethod to expose their info so that the generator phase could have a single method which figures out the tokens (and then moves this logic into the generator phase)

Other route would be to return a common struct or just have enough args on the function.

As note sure at the moment i like how this is effectively "generating" something in the parser phase :thinking:

Reply:
@LeonMatthesKDAB oops wrong account ^^ :sweat_smile:

Reply:
Right, this probably should go in the generator :thinking: 

And I guess it could just take all the data it needs directly
```rust
fn generate_member_parameter_list(mutability: bool /*or some kind of mutability enum*/, class_name: &Ident, parameters: &[ParsedFunctionParameter])
```
Then calling it for the signal would just be:
```rust
generate_member_parameter_list(signal.mutable, format_ident!("..."), &signal.parameters);
// and basically the same for methods
generate_member_parameter_list(method.mutable, format_ident!("..."), &method.parameters);
```
Which would be simple enough I think.

Reply:
So go with just using a common function then? I think a trait could potentially be useful elsewhere since method and signal have so much in common but might be a bit much work here just to simplify generation

Reply:
Done